src/libraries/modulemetadataparser.cpp

```diff
@@ -43,35 +45,32 @@ void foreachItem(const QJsonValue &value, Func f)
         f(item);
 }
 
-QString readName(const QJsonObject &object)
+QString readNotEmptyStringField(const QJsonObject &object, const QString &fieldName)
 {
-    const auto name = object["name"].toString();
-    if (name.isEmpty())
-        throw ModuleMetadataParser::Error("'name' missing or empty");
-    return name;
+    const auto value = object[fieldName].toString();
+    if (value.isEmpty())
+        throw ModuleMetadataParser::Error("'" + fieldName.toStdString() + "' midding or empty");
+    return value;
 }
 
-QString readDescription(const QJsonObject &object)
+QString readOptionalStringField(const QJsonObject &object,
+                                const QString &fieldName,
+                                const QString &defaultValue)
 {
-    return object["description"].toString();
+    const auto value = object[fieldName].toString();
+    if (value.isEmpty())
+        return defaultValue;
+    return value;
 }
 
-QStringList toStringList(const QJsonValue &value)
+QString readName(const QJsonObject &object)
 {
-  QStringList result;
-  foreachItem(value, [&result](const QJsonValue &item) {
-      result.append(item.toString());
-    });
-  return result;
+    return readNotEmptyStringField(object, "name");
 }
 
-Metadata::Import readImport(const QJsonObject &object)
+QString readDescription(const QJsonObject &object)
 {
-    const auto import = Metadata::Import{object["from"].toString(),
-                                         toStringList(object["types"])};
-    if (import.from().isEmpty() || import.types().isEmpty())
-        throw ModuleMetadataParser::Error("'from' and 'types' must be present and not-empty");
-    return import;
+    return object["description"].toString();
```

Comment:
is there a reason, why readNotEmptyStringField() or readOptionalStringField() is not used?

Reply:
yep, it's just older code (created before those two new functions)

but true, I should refactor that